controllers/kserve_inferenceservice_controller_authconfig_test.go

```diff
@@ -0,0 +1,222 @@
+					g.Expect(getAuthConfig(namespace.Name, isvc.Name, ac)).To(Succeed())
+					g.Expect(ac.Spec.Authorization["kubernetes-user"]).NotTo(BeNil())
+				}).
+					WithTimeout(timeout).
+					WithPolling(interval).
+					Should(Succeed())
+
+				Expect(disableAuth(isvc)).To(Succeed())
+				Eventually(func(g Gomega) {
+					ac := &authorinov1beta2.AuthConfig{}
+					g.Expect(getAuthConfig(namespace.Name, isvc.Name, ac)).To(Succeed())
+					g.Expect(ac.Spec.Authorization["anonymous-access"]).NotTo(BeNil())
+				}).
+					WithTimeout(timeout).
+					WithPolling(interval).
+					Should(Succeed())
+			})
+		})
+	})
+})
+
+func getAuthConfig(namespace, name string, ac *authorinov1beta2.AuthConfig) error {
+	return cli.Get(context.Background(), types.NamespacedName{Namespace: namespace, Name: name}, ac)
+}
+
+func createISVCMissingStatus(namespace string) *kservev1beta1.InferenceService {
+	inferenceService := &kservev1beta1.InferenceService{}
+	err := convertToStructuredResource(KserveInferenceServicePath1, inferenceService)
+	Expect(err).NotTo(HaveOccurred())
+	inferenceService.Namespace = namespace
+	Expect(cli.Create(ctx, inferenceService)).Should(Succeed())
+	return inferenceService
+}
+
+func createISVCWithAuth(namespace string) *kservev1beta1.InferenceService {
+	inferenceService := createBasicISVC(namespace)
+	inferenceService.Annotations["enable-auth"] = "true"
+	Expect(cli.Create(ctx, inferenceService)).Should(Succeed())
+
+	// HMM... second routine to do updates?
+	url, _ := apis.ParseURL("http://iscv-" + namespace + "ns.apps.openshift.ai")
+	inferenceService.Status = kservev1beta1.InferenceServiceStatus{
+		URL: url,
+	}
+
+	return inferenceService
+}
+
+func createISVCWithoutAuth(namespace string) *kservev1beta1.InferenceService {
+	inferenceService := createBasicISVC(namespace)
+	Expect(cli.Create(ctx, inferenceService)).Should(Succeed())
+
+	// HMM... second routine to do updates?
+	url, _ := apis.ParseURL("http://iscv-" + namespace + "ns.apps.openshift.ai")
+	inferenceService.Status = kservev1beta1.InferenceServiceStatus{
+		URL: url,
+	}
+
+	return inferenceService
+}
+
+func createBasicISVC(namespace string) *kservev1beta1.InferenceService {
+	inferenceService := &kservev1beta1.InferenceService{}
+	err := convertToStructuredResource(KserveInferenceServicePath1, inferenceService)
+	Expect(err).NotTo(HaveOccurred())
+	inferenceService.Namespace = namespace
+	if inferenceService.Annotations == nil {
+		inferenceService.Annotations = map[string]string{}
+	}
+	return inferenceService
+}
+
+func updateISVCStatus(isvc *kservev1beta1.InferenceService) error {
+	url, _ := apis.ParseURL("http://iscv-" + isvc.Namespace + "ns.apps.openshift.ai")
+	isvc.Status = kservev1beta1.InferenceServiceStatus{
+		URL: url,
+	}
+	return cli.Status().Update(context.Background(), isvc)
+}
+
+func disableAuth(isvc *kservev1beta1.InferenceService) error {
+	delete(isvc.Annotations, "enable-auth")
+	delete(isvc.Annotations, "security.opendatahub.io/enable-auth")
```

Comment:
Maybe, these strings should be a `const`.

Reply:
Fixed https://github.com/opendatahub-io/odh-model-controller/pull/130/commits/a6c333a99c31f82e5f6a35d314bee22de3951685

Seeing them side by side I wonder if maybe we should have called it something like `security.opendatahub.io/authorization` instead...?

Reply:
It could be. Although, for enabling and disabling, I'm not sure if it is that useful to make distinction between authorization and authentication.

But you tell, I let you do the renaming if you think it is better.

Reply:
I was more thinking the short vs "auth" vs "auth..." and "auth-x" vs "x-auth" order. (not authentication vs authorization)

security.opendatahub.io/authorization-group
security.opendatahub.io/enable-auth

security.opendatahub.io/authorization-group
security.opendatahub.io/authorization-enabled

Reply:
The `security.opendatahub.io/authorization-enabled` is the one that I see as potentially noisy for _later_. Right now, we are dealing only with authZ and the naming looks fine. But once we start looking at proper SSO, we may want to expand this with the authN part and add back to the `AuthConfig` the redirect that we removed previously. Then, this annotation would work to enable/disable both authN and authZ together, and its name will no longer fully reflect what it is doing.

Reply:
is `security....io/enabled` sufficient? It hints at securing the Model. Would there be other types of 'security' that could make it confusing in the future at some point?

Reply:
> is security....io/enable sufficient?

Uhm... I'm OK with it, but...

> Would there be other types of 'security' 

I don't know....
The only other thing that has been a topic lately is around TLS certs. But IDK if we will use annotations or labels for that.